src/test/scala/io/iohk/ethereum/mpt/MerklePatriciaTrieSuite.scala

```diff
@@ -554,22 +554,25 @@ class MerklePatriciaTrieSuite extends AnyFunSuite with ScalaCheckPropertyChecks
     assert(proof.isEmpty)
   }
 
-  test("getProof returns empty result for non-existing key") {
-    forAll(keyValueListGen()) { keyValueList: Seq[(Int, Int)] =>
-      val input: Seq[(Array[Byte], Array[Byte])] = keyValueList
-        .map { case (k, v) => k.toString.getBytes() -> v.toString.getBytes() }
-
-      val trie = input
-        .foldLeft(emptyMpt) { case (recTrie, (key, value)) =>
-          recTrie.put(key, value)
-        }
-
-      input.toList.foreach(x => {
-        val keyToFind = x._1 ++ Array(Byte.MaxValue)
-        val proof = trie.getProof(keyToFind)
-        assert(proof.isEmpty)
-      })
-    }
+  test("getProof returns proof result for non-existing address") {
+      // given
+      val EmptyTrie = MerklePatriciaTrie[Array[Byte], Array[Byte]](emptyEphemNodeStorage)
+      val key1: Array[Byte] = Hex.decode("10000001")
+      val key2: Array[Byte] = Hex.decode("10000002")
+      val key3: Array[Byte] = Hex.decode("30000003")
+      val key4: Array[Byte] = Hex.decode("10000004") //a key that doesn't have a corresponding value in the trie
+
+      val val1: Array[Byte] = Hex.decode("0101")
+      val val2: Array[Byte] = Hex.decode("0102")
+      val val3: Array[Byte] = Hex.decode("0103")
+      val trie = EmptyTrie
+        .put(key1, val1)
+        .put(key2, val2)
+        .put(key3, val3)
+      // when
+      val proof: Option[Vector[MptNode]] = trie.getProof(key4)
+      // then
+      assert(proof.isDefined)
```

Comment:
Got confused (this test actually IS testing a negative scenario)
Maybe we should also check that vector is empty?

Reply:
the vector shouldn't be empty, it should contain the nodes it traversed while searching for the account. Butt good point to check not only that it's defined, but also that it contains something

Reply:
what does the case Some(Vector.empty) mean?

Reply:
it means there was no root node in MPT, so no proof gathered

Reply:
and what's the difference of meaning between None and Some(Vector.empty)?

Reply:
According to the spec and geth implementation - there always should be a proof in response, even if the proof is empty we should return the key (for which the search was requested), the value that defaults to zero and the proof, which defaults to empty Seq. That's why the getProof method on MerklePatriciaTrie can return None as a proof, but when we construct an object of StorageValueProof, we need an empty sequence instead of None

Reply:
ah, wait, I probably confused you with my answer :)
If you were asking about specifically MerkleParticiaTrie#getProof - case Some(Vector.empty) should never happen. If there's a root node - it will never be empty. If there's no root node - it's gonna return None. 

I was talking about the next level of transforming the results.

Reply:
Can this be resolved?

Reply:
nitpick: the signature could be `Option[NonEmptyList]` to remove the option of having a Some(empty) result.